kasa/klapprotocol.py

```diff
@@ -0,0 +1,477 @@
+                    f"Unable to connect to the device, timed out: {self.host}: {ex}"
+                )
+            except AuthenticationException as auex:
+                _LOGGER.debug("Unable to authenticate with %s, not retrying", self.host)
+                raise auex
+            except Exception as ex:
+                if retry >= retry_count:
+                    _LOGGER.debug("Giving up on %s after %s retries", self.host, retry)
+                    raise SmartDeviceException(
+                        f"Unable to connect to the device: {self.host}: {ex}"
+                    )
+                continue
+
+        # make mypy happy, this should never be reached..
+        raise SmartDeviceException("Query reached somehow to unreachable")
+
+    async def _execute_query(self, request: str, retry_count: int) -> Dict:
+        timeout = aiohttp.ClientTimeout(total=self.timeout)
+
+        async with aiohttp.ClientSession(
+            cookie_jar=self.jar, timeout=timeout
+        ) as session:
+            if not self.handshake_done or self.handshake_session_expired():
+                try:
+                    await self.perform_handshake(session)
+
+                except AuthenticationException as auex:
+                    _LOGGER.debug(
+                        "Unable to complete handshake for device %s, authentication failed",
+                        self.host,
+                    )
+                    raise auex
+
+            # Check for mypy
+            if self.encryption_session is not None:
+                payload, seq = self.encryption_session.encrypt(request.encode())
+
+            url = f"http://{self.host}/app/request"
+
+            response_status, response_data = await self.session_post(
+                session, url, params={"seq": seq}, data=payload
+            )
+
+            msg = f"at {datetime.datetime.now()}.  Host is {self.host}, Retry count is {retry_count}, Sequence is {seq}, Response status is {response_status}, Request was {request}"
+            if response_status != 200:
+                if _LOGGER.isEnabledFor(logging.ERROR):
+                    _LOGGER.error("Query failed after succesful authentication " + msg)
+                # If we failed with a security error, force a new handshake next time.
+                if response_status == 403:
+                    self.handshake_done = False
+                    self.authentication_failed = True
+                    raise AuthenticationException(
+                        f"Got a security error from {self.host} after handshake completed {self.discovery_data}"
+                    )
+                else:
+                    raise SmartDeviceException(
+                        f"Device {self.host} responded with {response_status} to request with seq {seq}"
+                    )
+            else:
+                if _LOGGER.isEnabledFor(logging.DEBUG):
+                    _LOGGER.debug("Query posted " + msg)
```

Comment:
```suggestion
                _LOGGER.debug("Query posted %s", msg)
```
As `msg` is already created, using regular placeholders like this will not cause any overhead and there is no need to guard them.

Reply:
Done